lib/s3storage.php

```diff
@@ -126,10 +127,18 @@ public function writeObject($urn, $stream) {
 		}
 
 		$uploader = new ObjectUploader($this->connection, $this->getBucket(), $urn, $stream, 'private', $opt);
-		$uploader->upload();
+		$status = $uploader->upload();
+		$statusCode = $status->get('@metadata')['statusCode'];
 		if (\is_resource($stream)) {
 			\fclose($stream);
 		}
+
+		/**
+		 * Upload to object storage is successful when status code is 200
+		 */
+		if ($statusCode !== 200) {
+			throw new ObjectStoreWriteException("Something went wrong while writing the file. Kindly contact administrator", $statusCode);
```

Comment:
Any idea if you can get any error message from the uploader? Maybe `$status->get('@metadata')['errorMessage']` or something like that?
I expect the exception to be logged at some point, and having a clear message of what happened will make things easier to debug.

Reply:
Looked into the `@metadata` array and the return array from `upload()` method, couldn't find anything related to error message. Here is the list from `@metadata`:
```
'statusCode' => 200,
  'effectiveUri' => 'http://localhost:8000/owncloud/urn%3Aoid%3A18',
  'headers' => 
  array (
    'etag' => '"be0500fced8120f1f5d2471858f12059"',
    'x-amz-version-id' => '39383434323739333535343031323939393939395247303031202034',
    'server' => 'S3 Server',
    'x-amz-id-2' => '1371249df942b96c3ab0',
    'x-amz-request-id' => '1371249df942b96c3ab0',
    'date' => 'Tue, 07 May 2019 05:20:45 GMT',
    'connection' => 'close',
  ),
  'transferStats' => 
  array (
    'http' => 
    array (
      0 => 
      array (
      ),
    ),
  ),
```

Reply:
The examples I've seen catch an exception thrown by the uploader (https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/s3-multipart-upload.html and https://docs.aws.amazon.com/sdk-for-php/v3/developer-guide/getting-started_basic-usage.html).
If it works, it should be a better option in this case.

Reply:
Just check for the `AwsException`. I guess the intention for `MultipartUploadException` is to retry a chunk that has failed to be uploaded. We can skip this for now because this logic wasn't implemented, unless it's easy to implement. Note that the source stream might not be "rewindable", and also we cannot keep retrying the upload undefinitely.
For the status code 200, I'd prefer to skip it for now because it isn't shown in the examples. If the expectations fail we can try with it.

Reply:
Done. Removed the `MultipartUploadException` and the status code check for now in this PR as discussed. Right now the PR has only `AwsException`.